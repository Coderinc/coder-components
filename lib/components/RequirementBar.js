import React from 'react';
import styled from 'styled-components';
import Paragraph from '../elements/Paragraph';
import Chip from '../elements/Chip';

const BarContainer = styled.div`
  background-color: ${props =>
    props.active ? props.theme.activeBackground : props.theme.offsetBackground};
  color: ${props => props.theme.primaryFont};

  box-sizing: border-box;
  border-radius: 0.125rem;
  padding: 0.5rem;
  margin: 0.25rem 0;
  width: 100%;

  display: flex;

  &::before {
    font-size: 1.5rem;
    font-family: coder-icons;
    font-weight: normal;
    content: '\f108';
    margin-right: 0.5rem;
  }
`;

const BarContent = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

const BarTop = styled.div`
  display: flex;
`;

const BarBottom = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

const Icon = styled.div`
  margin: 0 0.25rem;
  &::before {
    font-size: 1.5rem;
    color: ${props => props.theme.offsetFont};
    font-family: coder-icons;
    font-weight: normal;
    content: '${props => props.content}';
  }
`;

const Fill = styled.div`
  flex: 1;
`;

export const RequirementBar = ({ phases = [], label, ...props }) => (
  <BarContainer {...props}>
    <BarContent>
      <BarTop>
        <Paragraph>{label}</Paragraph>
        <Fill />
        <Icon content="\f107" />
        <Icon content="\f11f" />
      </BarTop>
      <BarBottom>{phases.map(p => (<Chip
        key={p}
        label={p}
      />))}</BarBottom>
    </BarContent>
  </BarContainer>
);

export default RequirementBar;
