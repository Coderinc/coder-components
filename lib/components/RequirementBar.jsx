import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Paragraph from '../elements/Paragraph';
import Chip from '../elements/Chip';

const BarContainer = styled.div`
  background-color: ${({ active, theme }) => (active ? theme.activeBackground : theme.offsetBackground)};
  color: ${({ theme }) => theme.primaryFont};

  box-sizing: border-box;
  border-radius: 0.125rem;
  padding: 0.5rem;
  margin: 0.25rem 0;
  width: 100%;

  display: flex;

  &::before {
    font-size: 1.5rem;
    font-family: coder-icons;
    font-weight: normal;
    content: '\f108';
    margin-right: 0.5rem;
  }
`;

const BarContent = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

const BarTop = styled.div`
  display: flex;
`;

const BarBottom = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

const Icon = styled.div`
  margin: 0 0.25rem;
  &::before {
    font-size: 1.5rem;
    color: ${({ theme }) => theme.offsetFont};
    font-family: coder-icons;
    font-weight: normal;
    content: '${({ content }) => content}';
  }
`;

const Fill = styled.div`
  flex: 1;
`;

const RequirementBar = ({ phases, label, ...props }) => (
  <BarContainer {...props}>
    <BarContent>
      <BarTop>
        <Paragraph>
          {label}
        </Paragraph>
        <Fill />
        <Icon content="\f107" />
        <Icon content="\f11f" />
      </BarTop>
      <BarBottom>
        {phases.map(p => <Chip key={p} label={p} />)}
      </BarBottom>
    </BarContent>
  </BarContainer>
);

RequirementBar.propTypes = {
  phases: PropTypes.arrayOf(PropTypes.string),
  label: PropTypes.string.isRequired,
};

RequirementBar.defaultProps = {
  phases: [],
};

export default RequirementBar;
