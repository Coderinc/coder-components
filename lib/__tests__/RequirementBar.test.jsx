import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import sinon from 'sinon';
import 'jest-styled-components';
import RequirementBar from '../components/RequirementBar';

describe('RequirementBar', () => {
  it('renders correctly', () => {
    const component = renderer.create(<RequirementBar label="Lorem Ipsum" phases={["Design"]} />)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('fires on click', () => {
    const onClick = sinon.spy();
    const component = mount(<RequirementBar label="Lorem" phases={["t"]} value={1} onClick={onClick} />);
    component.find('BarContainer').simulate('click');
    expect(onClick.calledOnce).toEqual(true);
  });
})