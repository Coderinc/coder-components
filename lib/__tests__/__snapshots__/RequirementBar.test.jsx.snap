// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequirementBar renders correctly 1`] = `
.c3 {
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: lighter;
}

.c8 {
  padding: 0.25rem 0.5rem;
  margin: 0.25rem;
  border-radius: 0.125rem;
  font-size: 0.625rem;
}

.c0 {
  box-sizing: border-box;
  border-radius: 0.125rem;
  padding: 0.5rem;
  margin: 0.25rem 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0::before {
  font-size: 1.5rem;
  font-family: coder-icons;
  font-weight: normal;
  content: '\\f108';
  margin-right: 0.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c5 {
  margin: 0 0.25rem;
}

.c5::before {
  font-size: 1.5rem;
  font-family: coder-icons;
  font-weight: normal;
  content: '\\f107';
}

.c6 {
  margin: 0 0.25rem;
}

.c6::before {
  font-size: 1.5rem;
  font-family: coder-icons;
  font-weight: normal;
  content: '\\f11f';
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <p
        className="c3"
      >
        Lorem Ipsum
      </p>
      <div
        className="c4"
      />
      <div
        className="c5"
        content="\\\\f107"
      />
      <div
        className="c6"
        content="\\\\f11f"
      />
    </div>
    <div
      className="c7"
    >
      <div
        className="c8"
        label="Design"
      >
        Design
      </div>
    </div>
  </div>
</div>
`;
